delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/wiring/DeltaModule.scala

```diff
@@ -62,7 +62,7 @@ class DeltaModule(appCfg: AppConfig, config: Config)(implicit classLoader: Class
   make[StrictEntity].from { appCfg.http.strictEntityTimeout }
   make[ServiceAccount].from { appCfg.serviceAccount.value }
 
-  implicit val scheduler: Scheduler = Scheduler.global
+  implicit val executionContext: ExecutionContext = ExecutionContext.global
```

Comment:
Think this would ideally be a small fixed threadpool , assuming everywhere it's used is for non-blocking IO (like the HTTP client). According to the [CE docs](https://typelevel.org/cats-effect/docs/2.x/concurrency/basics#choosing-thread-pool) the key thing is to have a small number of threads dedicated to handling the async notification (the "event dispatcher" pool) and immediately shift back to the optimised compute pool (the one from `IOApp`). I always go back to this gist - https://gist.github.com/djspiewak/46b543800958cf61af6efa8e072bfd5c

The problem with the global EC is we don't know what else is scheduled on it (e.g. by libraries) so it could add latency when these notifications aren't processed immediately.

Having said that, it's [apparently](https://typelevel.org/cats-effect/docs/2.x/concurrency/basics#asynchronous--semantic-blocking) fine to use the default CE compute pool for non-blocking HTTP clients like this. Note the big caveat that the HTTP client must be non-blocking, which Akka HTTP is I think. I'm going to ask in Discord, because I've always heard both recommendations and it feels like a bit of a contradiction 🤷 

Sorry, a bit long, just feels important because `HttpClient` is used everywhere under the hood. As the docs say, `when it comes to concurrency noone has any idea what they're doing` 😂

Reply:
My understanding was the `Scheduler.global` (the EC we were previously using) is the same as `ExecutionContext.global`, so I think my change doesn't make anything worse

Reply:
It would be better to user the execution context provided by akka that you can access via `actorSystem.dispatcher`
That is the option we took in other cases

Reply:
Beat me to it! [Docs](https://doc.akka.io/docs/akka-http/current/handling-blocking-operations-in-akka-http-routes.html). Just not sure what happens when going to IO